lib/cli-engine.js

```diff
@@ -77,28 +77,7 @@ var DEFAULT_PARSER = require("../conf/eslint.json").parser;
 // Private
 //------------------------------------------------------------------------------
 
-var defaultOptions = {
-    configFile: null,
-    baseConfig: false,
-    rulePaths: [],
-    useEslintrc: true,
-    envs: [],
-    globals: [],
-    rules: {},
-    extensions: [".js"],
-    ignore: true,
-    ignorePath: null,
-    parser: DEFAULT_PARSER,
-    cache: false,
-    // in order to honor the cacheFile option if specified
-    // this option should not have a default value otherwise
-    // it will always be used
-    cacheLocation: "",
-    cacheFile: ".eslintcache",
-    fix: false,
-    allowInlineConfig: true,
-    cwd: process.cwd()
-};
+defaultOptions = assign({}, defaultOptions, {cwd: process.cwd()});
```

Comment:
The problem is that the `cwd` in `defaultOptions` will be whatever it was when first required by any module.  But, I thought it would be safer to override it here, to maintain the same behavior it had before I extracted out the defaults.  I have to be very careful to override it in [other places](https://github.com/eslint/eslint/pull/4467/files#diff-d1aa1e3567e7e5b9cf9be2fbb4cab8cfR67)
 as well, or else I end up with the wrong cwd in my tests.  This is important because I am moving files into a temp folder in the tests, isolate them from the `.eslintignore` within eslint.  If I don't override the cwd, it ends up being the project root, and the project `.eslintignore` still causes the files to be ignored.  I'm not sure if this makes any sense, but in any event I figured it's better safe than sorry in this case you mention.